sdk/python/tests/unit/test_on_demand_feature_view.py

```diff
@@ -32,10 +38,18 @@ def udf1(features_df: pd.DataFrame) -> pd.DataFrame:
 def udf2(features_df: pd.DataFrame) -> pd.DataFrame:
     df = pd.DataFrame()
     df["output1"] = features_df["feature1"] + 100
-    df["output2"] = features_df["feature2"] + 100
+    df["output2"] = features_df["feature2"] + 101
     return df
 
 
+def python_native_udf(features_dict: Dict[str, List[Any]]) -> Dict[str, Any]:
```

Comment:
@franciscojavierarceo sorry for the late find. Can you look at this test examples? The signatures are correct, but actual transformation seems to assume that `features_dict["feature1"]` is an int, but it should be a list of ints, right?

Reply:
whoops let me update the test case tomorrow but it should be a singleton, the type hint is wrong. I adjusted this late and missed this one.

Reply:
ah, got it. If it should be a singleton, then also double-check what is actually passed from odfv class as well. I may be wrong, but I think it's passing a list.

Reply:
I'll add another test when using `get_online_features` for an ODFV to confirm it more explicitly. I do have a test below confirming the correct behavior for `self.feature_transformation.transform()` returning the expected result.

Reply:
I'm planning to add an explicit `get_online_features` test for `substrait` as well. Just testing `transform` is clearly not enough.

Reply:
here's the fix for the type: https://github.com/feast-dev/feast/pull/4054